rust/src/address.rs

```diff
@@ -165,21 +200,22 @@ impl ByronAddress {
         self.0.attributes.serialize(&mut attributes_bytes).unwrap();
         attributes_bytes.finalize()
     }
-    pub fn network_id(&self) -> u8 {
+    pub fn network_id(&self) -> Result<u8, JsValue> {
         // premise: during the Byron-era, we had one mainnet (764824073) and many many testnets
         // with each testnet getting a different protocol magic
         // in Shelley, this changes so that:
         // 1) all testnets use the same u8 protocol magic
         // 2) mainnet is re-mapped to a single u8 protocol magic
 
         // recall: in Byron mainnet, the network_id is omitted from the address to save a few bytes
-        let mainnet_network_id = 764824073;
         // so here we return the mainnet id if none is found in the address
 
-        match self.0.attributes.network_magic {
-            // although mainnet should never be explicitly added, we check for it just in case
-            Some(x) => if x == mainnet_network_id { 0b0001 } else { 0b000 },
-            None => 0b0001, // mainnet is implied if omitted
+        // mainnet is implied if omitted
+        let protocol_magic = self.0.attributes.network_magic.unwrap_or(NetworkInfo::mainnet().protocol_magic());
```

Comment:
What if we just used `let protocol_magic = self.0byron_protocol_magic()` here since that logic is already done above?